TACC/Core-Portal
# FP-1491: Style _common Button React Component

## Overview: ##

Add styles and tweaks to Button component.

## Related Jira tickets: ##

* [FP-1491](https://jira.tacc.utexas.edu/browse/FP-1491)
* mirrored by https://github.com/TACC/Core-Styles/pull/20

## Summary of Changes: ##

For button component:
- Demonstrate new styles _only_ in UI Patterns.
- Explain and show more Buttons.
- Migrate styles to `styles/components/c-button--new.css`.
- Set limitations and logging for component props.

## UI Photos:

<details open><summary>After Fixes from Review</summary>

![FP-1491 Top](https://user-images.githubusercontent.com/62723358/167716297-d812459c-9dbf-4569-8f14-105be3f6dc12.png)
![FP-1491 Bottom](https://user-images.githubusercontent.com/62723358/167716299-648043f1-9d15-43fc-b5e0-ad3a48a95d37.png)

https://user-images.githubusercontent.com/62723358/168176800-39da10ef-2826-439d-a7c2-37ddd4af090e.mov

</details>
<details><summary>Before Fixes from Review</summary>

![FP-1491 UI Patterns Top](https://user-images.githubusercontent.com/62723358/166324828-e3b88804-9d82-4eb0-9b35-b6adbf139e40.png)
![FP-1491 UI Patterns Bottom](https://user-images.githubusercontent.com/62723358/166324823-5ff2ad66-9f58-44e4-a881-05acd22fcb91.png)

https://user-images.githubusercontent.com/62723358/166325195-42787f92-3e32-4dbf-a8ca-cf7258c6b118.mov

</details>



## Testing Steps: ##

1. Visit UI Patterns.
2. Scroll to Buttons.
3. Manually test for oddities.
4. Verify styles and interactions match the design.
5. Surf the rest of the app to verify button styles are unchanged.[^1]

[^1]: App is not supposed to use new styles until [FP-546](https://jira.tacc.utexas.edu/browse/FP-546).



## Notes: ##

### Design

- [CMS & Portal (official)](https://xd.adobe.com/view/b37b291a-7798-4814-90f3-7ac63c48a3e0-e628/screen/d7a72466-4ae8-43fd-9531-44e033245ddb) (the right two columns are for Portal)
- [Portal (screenshot)](https://user-images.githubusercontent.com/62723358/166328046-7b813b65-9d9a-48c8-9c07-5a558177a398.png)
